src/language-markdown/printer-markdown.js

```diff
@@ -298,7 +298,9 @@ function genericPrint(path, options, print) {
         ":",
         indent([
           lineOrSpace,
-          printUrl(node.url),
+          options.parser !== "mdx" && node.url === ""
+            ? "<>"
+            : printUrl(node.url),
```

Comment:
Other places where URLs are printed don't require `<>` for empty URL. `[image]()` and `[link][]` don't need `<>`.